early-return/workflow_test.go

```diff
@@ -0,0 +1,102 @@
+package earlyreturn
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/sdk/testsuite"
+)
+
+func Test_CompleteTransaction(t *testing.T) {
+	testSuite := &testsuite.WorkflowTestSuite{}
+	env := testSuite.NewTestWorkflowEnvironment()
+
+	tx := Transaction{ID: uuid.New(), SourceAccount: "Bob", TargetAccount: "Alice", Amount: 100}
+	env.RegisterActivity(tx.InitTransaction)
+	env.RegisterActivity(tx.CompleteTransaction)
+
+	uc := &updateCallback{}
+	env.RegisterDelayedCallback(func() {
```

Comment:
nit: would add a comment explaining this will guarantee the update is sent in the first WFT.

Reply:
FWIW I'm confused even after Quinn's comment.  Why is this test seemingly not using the normal interface for UwS?
(I assume you have a good reason and that there's nothing fixable in this PR.)

Reply:
Is it not backed by the Java test service?

Reply:
Correct; Go SDK has it's own time-skipping test server, and it has its own APIs. We haven't added an UwS API since this approach here works, too.